pkg/keyidentifier/keyidentifier_test.go

```diff
@@ -13,15 +13,26 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+type spec struct {
+	KeyPath                   string
+	ExpectedFingerprintSHA256 string
+	ExpectedFingerprintMD5    string
+	Encrypted                 bool
+	Bits                      int
+	Type                      string
+	Format                    string
+	Source                    string
+}
+
 // TestIdentifyFiles walks the testdata directory, and tests each
-// file. The expected results are extrapolated from the directory path
+// file.
 func TestIdentifyFiles(t *testing.T) {
-	kIdentifer, err := New(WithLogger(logutil.NewCLILogger(true)))
+	kIdentifier, err := New(WithLogger(logutil.NewCLILogger(true)))
 	require.NoError(t, err)
 
 	testFiles := []string{}
 
-	err = filepath.Walk("testdata", func(path string, info os.FileInfo, err error) error {
+	err = filepath.Walk("testdata/specs", func(path string, info os.FileInfo, err error) error {
```

Comment:
This is now a single level, you can get simpler than Walk. glob for json is probably simple?